File: Year-2023/LeetCode-November-Challenge/DetermineifaCellIsReachableataGivenTime.java

class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int xDiff = Math.abs(fx - sx), yDiff = Math.abs(fy - sy);
        if(xDiff == 0 && yDiff == 0 && t == 1) return false;
        return Math.min(xDiff, yDiff) + Math.abs(xDiff - yDiff) <= t;
    }
}

public class DetermineifaCellIsReachableataGivenTime{
    public static void main(String[] args) {
        int sx = 2, sy = 4, fx = 7, fy = 7, t = 6;
        Solution s = new Solution();
        System.out.println(s.isReachableAtTime(sx, sy, fx, fy, t));
    }
}

File: Year-2023/LeetCode-November-Challenge/DetermineifaCellIsReachableataGivenTime.js
var isReachableAtTime = function (sx, sy, fx, fy, t) {
    let xDff = Math.abs(fx - sx), yDiff = Math.abs(fy - sy);
    if (xDff === 0 && yDiff === 0 && t === 1) return false;
    return Math.min(xDff, yDiff) + Math.abs(xDff - yDiff) <= t;
};

let sx = 2, sy = 4, fx = 7, fy = 7, t = 6;
console.log(isReachableAtTime(sx, sy, fx, fy, t));